activation/activation_errors.go

```diff
@@ -60,3 +60,9 @@ func (e *PoetRegistrationMismatchError) Error() string {
 	}
 	return sb.String()
 }
+
+func (e *PoetRegistrationMismatchError) Is(target error) bool {
+	var poetRegistrationMismatchError *PoetRegistrationMismatchError
+	ok := errors.As(target, &poetRegistrationMismatchError)
+	return ok
+}
```

Comment:
This isn't correct. `errors.Is` compares value equality, `errors.As` type equality.

```go
err1 := &PoetRegistrationMismatchError{
    registrations: []string{"url1"},
}
err2 := &PoetRegistrationMismatchError{
    registrations: []string{"url2"},
}

fmt.Println(errors.Is(err1, err2)) // returns true although it should be false

Reply:
Done same like here 
https://github.com/spacemeshos/go-spacemesh/blob/develop/activation/activation_errors.go#L32

Can fix both of them.

Reply:
Yes, in my opinion this is not correct, but it also doesn't need to be fixed in this PR 🙂

Reply:
This PR fixes usage of `PoetSvcUnstableError`: https://github.com/spacemeshos/go-spacemesh/pull/6217